qiskit_nature/second_q/operators/majorana_op.py

```diff
@@ -0,0 +1,521 @@
+    copy, the dictionary will be stored by reference if you disable ``copy`` like so:
+
+    .. code-block:: python
+
+        some_big_data = {
+            "_0 _1": .25j,
+            "_1 _0": -.25j,
+            # ...
+        }
+
+        op = MajoranaOp(
+            some_big_data,
+            num_spin_orbitals=2,
+            copy=False,
+        )
+
+
+    .. note::
+
+        It is the users' responsibility, that in the above scenario, :code:`some_big_data` is not
+        changed after initialization of the ``MajoranaOp``, since the operator contents are not
+        guaranteed to remain unaffected by such changes.
+
+    **Construction from Fermionic operator**
+
+    The default way to construct a ``MajoranaOp`` is from an existing ``FermionicOp``:
+
+    .. code-block:: python
+
+        from qiskit_nature.second_q.operators import FermionicOp, MajoranaOp
+        f_op = FermionicOp({"+_0 -_1": 1}, num_spin_orbitals=2)
+        m_op = MajoranaOp.from_fermionic_op(f_op)
+
+    Note that every term of the ``FermionicOp`` will result in :math:`2^n` terms in the
+    ``MajoranaOp``, where :math:`n` is the number of fermionic modes in the term. The conversion
+    uses the convention that
+
+    .. math::
+
+        a_i = \frac{1}{2}(\gamma_{2i} + i \gamma_{2i+1}), \quad
+        a_i^\dagger = \frac{1}{2}(\gamma_{2i} - i \gamma_{2i+1}) \,,
+
+    where :math:`a_i` and :math:`a_i^\dagger` are the Fermionic annihilation and creation operators
+    and :math:`\gamma_i` the Majorana operators.
+
+    .. note::
+
+        When creating a ``MajoranaOp`` from a ``PolynomialTensor`` using
+        :meth:`from_polynomial_tensor`, the underscore character :code:`_` is the only allowed
+        character in the keys of the ``PolynomialTensor``.
+
+    **Algebra**
+
+    This class supports the following basic arithmetic operations: addition, subtraction, scalar
+    multiplication, operator multiplication, and adjoint.
+    For example,
+
+    Addition
+
+    .. code-block:: python
+
+      MajoranaOp({"_1": 1}, num_spin_orbitals=2) + MajoranaOp({"_0": 1}, num_spin_orbitals=2)
+
+    Sum
+
+    .. code-block:: python
+
+      sum(MajoranaOp({label: 1}, num_spin_orbitals=3) for label in ["_0", "_1", "_2 _3"])
+
+    Scalar multiplication
+
+    .. code-block:: python
+
+      0.5 * MajoranaOp({"_1": 1}, num_spin_orbitals=2)
+
+    Operator multiplication
+
+    .. code-block:: python
+
+      op1 = MajoranaOp({"_0 _1": 1}, num_spin_orbitals=2)
+      op2 = MajoranaOp({"_0 _1 _2": 1}, num_spin_orbitals=2)
+      print(op1 @ op2)
+
+    Tensor multiplication
+
+    .. code-block:: python
+
+      op = MajoranaOp({"_0 _1": 1}, num_spin_orbitals=2)
+      print(op ^ op)
+
+    Adjoint
+
+    .. code-block:: python
+
+      MajoranaOp({"_0 _1": 1j}, num_spin_orbitals=2).adjoint()
+
+    .. note::
+
+        Since Majorana generators are self-adjoined, the adjoint of a ``MajoranaOp`` is the original
```

Comment:
Why do you call them "generators" now? We should be consistent with our wording here.

Reply:
indeed. changed.